src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,29 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+interface Props {
+  activeTodo: Todo;
+  setActiveTodo: (todo: Todo | null) => void;
+}
+
+export const TodoModal: React.FC<Props> = ({ activeTodo, setActiveTodo }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [todoUser, setTodoUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(activeTodo.userId)
+      .then(setTodoUser)
+      .finally(() => setIsLoading(false));
+  }, []);
```

Comment:
You have an empty array as a dependency for your `useEffect` hook. This means that the effect will only run once after the initial render. However, you're using `activeTodo.userId` inside the effect, which is a prop. If `activeTodo` changes, your effect won't run again as expected. You should include `activeTodo.userId` in the dependency array to ensure that the effect runs whenever `activeTodo.userId` changes.